2_process/src/match_sites_reaches.R

```diff
@@ -78,3 +80,71 @@ get_site_flowlines <- function(reach_sf, sites, sites_crs, max_matches = 1, sear
 }
 
 
+
+
+#' @description Function to match reaches (NHDv2 flowlines) to point sites 
+#' 
+#' @param sites sf data frame containing site locations
+#' @param sites_crs the crs of the sites table (i.e., 4269 for NAD83)
+#' @param nhd_lines sf data frame containing NHDPlusV2 flowlines
+#' @param max_matches the maximum number of segments that a point can match to
+#' @param search_radius the maximum radius in meters to match a point to a segment;
+#' segments outside of search_radius will not match
+#'
+#' @value A data frame that contains the same columns as the `sites` data,
+#' with additional columns `COMID`, `bird_dist_to_comid_m` where 
+#' `bird_dist_to_comid_m` is the distance (in meters) between the site and the
+#' matched NHDv2 flowline.
+#'
+get_site_nhd_flowlines <- function(nhd_lines, sites, sites_crs,
```

Comment:
Why did you decide to make a new function, `get_site_nhd_flowlines` and not use `get_site_flowlines`? The two look very similar.

Reply:
I made a new function because originally I wasn't going to delete the equivalent PRMS/NHM target, `p2_sites_w_segs` (at least not in this PR). These two functions are nearly identical aside from formatting that was needed to adapt the nhdplusTools function to the NHM fabric. We can delete `get_site_flowlines` if we're opting to remove all the NHM stuff from the pipeline. Should I go ahead and do that in this PR?